dask/layers.py

```diff
@@ -409,16 +388,23 @@ def __init__(
         # much earlier.
         #
         # See https://github.com/dask/dask/pull/6051 for a detailed discussion.
-        self.annotations = self.annotations or {}
-        if "priority" not in self.annotations:
-            self.annotations["priority"] = {}
-        self.annotations["priority"]["__expanded_annotations__"] = None
-        self.annotations["priority"].update({_key: 1 for _key in self.get_split_keys()})
+        annotations = annotations or {}
+        self._split_keys = None
+        if "priority" not in annotations:
+            annotations["priority"] = self._key_priority
+
+        self._split_keys_set = set(self.get_split_keys())
+        super().__init__(annotations=annotations)
+
+    def _key_priority(self, key):
+        if key in self._split_keys_set:
+            return 1
+        return 0
 
     def get_split_keys(self):
         # Return SimpleShuffleLayer "split" keys
         return [
-            stringify((self.split_name, part_out, part_in))
+            (self.split_name, part_out, part_in)
```

Comment:
💢 😡 

callables are evaluated on the actual, non-stringified keys. If this is a sane behavior or not is something I cannot truly judge but this is how it's been before. This is changing now because the `__expanded_annotations__` sentinel no longer exists.

Reply:
Evaluation on non-stringified keys makes sense to me.  For example we might want to prioritize based on the index number `("read-parquet", 15)` -> `15`